### DISCUSSION TITLE
How to get an entity's ->original entity without triggering "Access to an undefined property"

### DISCUSSION THREAD

**Discussion by alberto56** [mglaman/phpstan-drupal#612] [General]
When a node is in the process of being saved, an entity's ->original property will contain null if the entity is new, or the original entity if the entity is being changed.

However if we have code like:

```
...
$entity->original;
...
```

We get the error:

```
  78     Access to an undefined property                                           
         Drupal\Core\Entity\EntityInterface::$original.                            
         💡 Learn more:                                                            
            https://phpstan.org/blog/solving-phpstan-access-to-undefined-property  
```

The only way I could find to get rid of this error is to add:

```
    // @phpstan-ignore-next-line
```

Is there a better way to access the original property without triggering this error?

**Comment by dpi** [comment]
Consider running the patch ;) https://www.drupal.org/project/drupal/issues/2839195

**Comment by mglaman** [comment]
Or, add `@property EntityInterface|null $original` to https://github.com/mglaman/phpstan-drupal/blob/main/stubs/Drupal/Core/Entity/EntityInterface.stub
